plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java

```diff
@@ -0,0 +1,218 @@
+package org.jenkinsci.plugins.structs.describable;
+
+import hudson.model.Describable;
+import org.jenkinsci.Symbol;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+/**
+ * A reflection-like view of a particular {@link Describable} instance.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class UninstantiatedDescribable implements Serializable {
+    private String symbol;
+    private String klass;
+    private final Map<String,?> arguments;
+    private transient DescribableModel model;
```

Comment:
Not sure I follow. There are several methods that behave differently if `model` is unset. How can you be sure that they will never be called on a deserialized instance?

And was there actually a test somewhere that demonstrated instantiation after restart? If so, I must have missed it.